program/src/oracle/oracle.c

```diff
@@ -561,9 +561,21 @@ static uint64_t upd_price( SolParameters *prm, SolAccountInfo *ka )
 
   // update component price if required
   if ( cptr->cmd_ == e_cmd_upd_price || cptr->cmd_ == e_cmd_upd_price_no_fail_on_error ) {
+    uint32_t status = cptr->status_;
+
+    // Set publisher's status to unknown unless their CI is sufficiently tight.
+    int64_t threshold_conf = (cptr->price_ / PC_MAX_CI_DIVISOR);
+    if (threshold_conf < 0) {
+      // Safe as long as threshold_conf isn't the min int64, which it isn't as long as PC_MAX_CI_DIVISOR > 1.
+      threshold_conf = -threshold_conf;
+    }
```

Comment:
I believe this is equivalent to casting the result of the division to an unsigned integer.

```suggestion
    // Safe as long as threshold_conf isn't the min int64, which it isn't as long as PC_MAX_CI_DIVISOR > 1.
    uint64_t threshold_conf = (uint64_t)(cptr->price_ / PC_MAX_CI_DIVISOR);
```